nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -345,9 +373,47 @@ geometry_msgs::msg::PoseStamped RegulatedPurePursuitController::getLookAheadPoin
   return *goal_pose_it;
 }
 
+geometry_msgs::msg::PoseStamped
+RegulatedPurePursuitController::projectCarrotPastGoal(
+  const double & lookahead_dist, const nav_msgs::msg::Path & transformed_plan)
+{
+  if (transformed_plan.poses.size() == 1) {
+    // We cannot recover the direction of the last segment
+    // Put the lookahead point directly in front of the robot so that we don't
+    // turn
+    auto goal_pose = transformed_plan.poses.back();
+    geometry_msgs::msg::PoseStamped pose;
+    pose.header.frame_id = goal_pose.header.frame_id;
+    pose.header.stamp = goal_pose.header.stamp;
+    pose.pose.position.x =
+      goal_pose.pose.position.x > 0.0 ? lookahead_dist : -lookahead_dist;
+    pose.pose.position.y = 0.0;
+    return pose;
+  }
+
+  // Calculate the direction of the last segment
+  const auto goal_pose = transformed_plan.poses.end()[-1];
+  const auto prev_pose = transformed_plan.poses.end()[-2];
+  const auto dx = goal_pose.pose.position.x - prev_pose.pose.position.x;
+  const auto dy = goal_pose.pose.position.y - prev_pose.pose.position.y;
+  const auto d = std::hypot(dx, dy);
+
+  // Create a carrot the lookahead distance from the goal pose
+  auto pose = goal_pose;
+  pose.pose.position.x += dx / d * lookahead_dist;
+  pose.pose.position.y += dy / d * lookahead_dist;
+
+  // Interpolate as usual
+  auto point = circleSegmentIntersection(
+    prev_pose.pose.position, goal_pose.pose.position, lookahead_dist);
```

Comment:
Is this correct?, or should it be...
`circleSegmentIntersection(prev_pose.pose.position, pose.pose.position, lookahead_dist);`

Reply:
You are correct. Changed.

(Tests passed because the circle segment intersection calculation will calculate an intersection for a segment projected past the end point anyway)